api/application.ts

```diff
@@ -0,0 +1,185 @@
+import * as bodyParser from 'body-parser'
+import * as cookieParser from 'cookie-parser'
+import * as express from 'express'
+import * as session from 'express-session'
+import * as helmet from 'helmet'
+import * as log4js from 'log4js'
+import * as auth from './auth'
+import {environmentCheckText, getConfigValue, getEnvironment, showFeature} from './configuration'
+import {ERROR_NODE_CONFIG_ENV} from './configuration/constants'
+import {
+  APP_INSIGHTS_KEY,
+  COOKIE_TOKEN,
+  COOKIES_USERID,
+  FEATURE_APP_INSIGHTS_ENABLED,
+  FEATURE_HELMET_ENABLED,
+  FEATURE_PROXY_ENABLED,
+  FEATURE_SECURE_COOKIE_ENABLED,
+  HELMET,
+  IDAM_CLIENT,
+  IDAM_SECRET,
+  LOGGING,
+  MAX_LINES,
+  MAX_LOG_LINE,
+  MICROSERVICE,
+  NOW,
+  OAUTH_CALLBACK_URL,
+  PROTOCOL,
+  S2S_SECRET,
+  SERVICE_S2S_PATH,
+  SERVICES_IDAM_API_PATH,
+  SERVICES_IDAM_WEB,
+  SERVICES_RD_PROFESSIONAL_API_PATH,
+  SESSION_SECRET,
+} from './configuration/references'
+import {appInsights} from './lib/appInsights'
+import {errorStack} from './lib/errorStack'
+import {getStore} from './lib/sessionStore'
+import * as tunnel from './lib/tunnel'
+import routes from './routes'
+import serviceRouter from './services/serviceAuth'
+
+export const app = express()
+
+export const logger = log4js.getLogger('server')
+logger.level = getConfigValue(LOGGING) ? getConfigValue(LOGGING) : 'OFF'
+
+/**
+ * If there are no configuration properties found we highlight this to the person attempting to initialise
+ * this application.
+ */
+if (!getEnvironment()) {
+  console.log(ERROR_NODE_CONFIG_ENV)
+}
+
+/**
+ * TODO: Implement a logger on the Node layer.
+ */
+console.log(environmentCheckText())
+
+// TODO: Testing that we can get the environment variables on AAT from the .yaml file
+console.log('ENV PRINT')
```

Comment:
I added these, but once we get up to production, when AKS is complete we'll have to remove these.

Reply:
@ficklephil  I will remove them after it goes through AAT